docs/src/extending.md

```diff
@@ -26,16 +26,42 @@ Base.close(connection::MyConnection) = (connection.isopen = false)
 open!(connection::MyConnection) = (connection.isopen = true)
 
 function setup_connection(session::Session{MyConnection})
+    # If you need to do something like start an HTTP server, you can do it here, synchronously.
+
     return js"""
-    // Javascript needed to connect to
-    const conn = create_connection(...) // implemented by your framework
-    conn.on_msg((msg) => {
-        JSServe.process_message(msg)
-    });
-    // register sending message
-    JSServe.on_connection_open((binary) => {
-        comm.send(binary)
-    }, $(session.compression_enabled));
+    // TODO: create a connection
+    create_connection(...).then((conn) => {
+        
+        // TODO: when your connection receives a message from Julia, relay the message to `JSServe.process_message(msg)`.
+        conn.on_msg((msg) => {
+            JSServe.process_message(msg)
+        });
+        
+        // TODO: you need to define a JavaScript function that sends a given `binary_data` to Julia. On the Julia side, this should call `JSServe.process_message(connection.parent, binary_data)`.
+        const send_to_julia = (binary_data) => conn.send(binary)
+        
+        // TODO: does your connection use Pako compression on its incoming and outgoing messages?
+        const compression_enabled = false
+        
+        // Register your new connection
+        JSServe.on_connection_open(send_to_julia, compression_enabled);
+    })
+    
     """
 end
 ```
+
+
+You can test your new connection like so:
+
+```Julia
+JSServe.register_connection!(MyConnection) do
+    if i_want_to_use_it
+        # TODO: should I create a new session myself?
+        session = Session()
+        return MyConnection(session)
```

Comment:
You don't need to create a session here, this will get created automatically! A connection is not in control of a session, instead a Session creates a connection to communicate.